src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -440,12 +440,6 @@ void InlineTypeNode::load(GraphKit* kit, Node* base, Node* ptr, ciInstanceKlass*
           ft = con_type->inline_klass();
           null_free = true;
         }
-        BasicType bt = con_type->basic_type();
-        int vec_len = field->secondary_fields_count();
-        if (field->is_multifield_base() &&
-          Matcher::match_rule_supported_vector(VectorNode::replicate_opcode(bt), vec_len, bt)) {
-          value = kit->gvn().transform(VectorNode::scalar2vector(value, vec_len, Type::get_const_type(field->type()), false));
-        }
```

Comment:
The reason that I change this here is I met another issue when the constant oops is `IOTA` which saves the index values. For such cases, the `scalar2vector` will result into wrong result. Besides, for subtype vectors, the basic type from line-443 (i.e. `BasicType bt = con_type->basic_type();`) is promoted to `int`. I cannot see the issue if loading a vector from a constant oop.  Could you please give an example? Alternatively, I was thinking about generating different IRs for the `ZERO` and `IOTA` constants. But we need to do more check about the actual values, which I think is not so urgent now.

Reply:
It will not be a problem with constant oops or even default values which are value_mirrors, but gvn.zerocon(T_PRIMITIVE_OBJECT) will be create a ConP node representing a NULL_PTR, any address commutation w.r.t. base is will result into illegal address.  

We should keep multi-field handling in both the control paths and add an extract check for ConP (TypePtr::NULL_PTR) in if condition for Type::NULL_PTR  it will handle all the cases.

Reply:
Do you have a test when the constant oop's type is NULL_PTR? I can prepare a fixing and testing. Thanks!

Reply:
I encountered this issue during my implementation when we attempt to create a VectorBoxNode while ciInlineKlass was not initialized.
https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/vectornode.hpp#L1692
You can make the change in above line to always create gvn.zerocon(T_PRIMITIVE_OBJECT) oop and remove special multifield handlings for constant case in InlineTypeNode::load, it will result into creation of imbalanced PhiNodes.

test Float256VectorTests.blendFloat256VectorTestsBroadcastLongSmokeTest(float[-i * 5], float[cornerCaseValue(i)], mask[false]): success

Error mixing types: vectory[8]:{float} and ftcon:0.000000

Reply:
Thanks! I will take a look at this issue.

Reply:
As per my understanding, oop from `gvn.zerocon(T_PRIMITIVE_OBJECT)` is a NULL_PTR, which should be handled in `InlineTypeNode::make_from_oop()` before calling `InlineTypeNode::load()` (see:  https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L857). Not sure why such issue happens.

Reply:
> As per my understanding, oop from `gvn.zerocon(T_PRIMITIVE_OBJECT)` is a NULL_PTR, which should be handled in `InlineTypeNode::make_from_oop()` before calling `InlineTypeNode::load()` (see: https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L857). Not sure why such issue happens.

VectorPayloadMF* classes are primitive classes and their instances are non-nullable.

Reply:
Make sense. But the condition in [line-854](https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L854) is checking the oop itself instead of the `null_free` input. So if the oop is the `NULL_PTR`, it will be catched as well? I tried to hard the oop to be `gvn.zerocon(T_PRIMITIVE_OBJECT)`, and the code goes to [line-859](https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L859). Hence the followed `load()` cannot be executed.  

BTW, did you run the test again with the latest patch of this PR?  Not sure whether the issue you met before is fixed, since we have changed a lot in compiler.  Maybe we can add an assertion that the constant oop is not `NULL_PTR`, WDYT?

Reply:
> Make sense. But the condition in [line-854](https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L854) is checking the oop itself instead of the `null_free` input. So if the oop is the `NULL_PTR`, it will be catched as well? I tried to hard the oop to be `gvn.zerocon(T_PRIMITIVE_OBJECT)`, and the code goes to [line-859](https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.cpp#L859). Hence the followed `load()` cannot be executed.
> 
Unbalanced PhiNode indicate creation of a scalar, you can put a FIXME over the specific code, we can then address it in subsequent patches.

> BTW, did you run the test again with the latest patch of this PR? Not sure whether the issue you met before is fixed, since we have changed a lot in compiler. Maybe we can add an assertion that the constant oop is not `NULL_PTR`, WDYT?

Thanks, did not do a test re-run with latest patch as there were several regressions in pervious one, one quick comment, kindly set can_optimize = false at following line,
https://github.com/openjdk/valhalla/pull/845/files#diff-e368502932198721c3ee52ace2397c5a9c6281930e83fc53a429573f6adf44fcR2584

I have bug fix patch ready over my pending merge patch with which I see 90% *VectorTests*.java test passing,  I shall float it once all tests are green.

Its ok to integrate your patch first, followed by a re-merge and then bug fixes for failing JTREG regression over it. WDYT ?

Reply:
> Its ok to integrate your patch first, followed by a re-merge and then bug fixes for failing JTREG regression over it. WDYT ?

Sounds good to me! Thanks!

Reply:
> Thanks, did not do a test re-run with latest patch as there were several regressions in pervious one, one quick comment, kindly set can_optimize = false before following line,
https://github.com/openjdk/valhalla/pull/845/files#diff-e368502932198721c3ee52ace2397c5a9c6281930e83fc53a429573f6adf44fcR2584

It's seems not right setting `can_optimize = false` before line-2584 for me. This will break the inner loop anyway, since it doesn't set `can_optimize = true` in the followed code, does it?

Reply:
> > Thanks, did not do a test re-run with latest patch as there were several regressions in pervious one, one quick comment, kindly set can_optimize = false before following line,
> > https://github.com/openjdk/valhalla/pull/845/files#diff-e368502932198721c3ee52ace2397c5a9c6281930e83fc53a429573f6adf44fcR2584
> 
> It seems not right setting `can_optimize = false` before line-2584 for me. This will break the inner loop anyway, since it doesn't set `can_optimize = true` in the followed code, does it?

I can see an incorrect graph shape getting generated for ABSMaskedByte256VectorTest

```
2970  VectorBox  === _ 2967 34 2969  [[ 562 557 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !orig=[3040],... !jvms: VectorMask::fromArray @ bci:47 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
  561  IfTrue  === 559  [[ 2883 562 ]] #1 !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 2883  If  === 561 2882  [[ 2884 2885 ]] P=0.000001, C=-1.000000 !orig=[584] !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1181  LoadVector  === 2876 2991 1180  [[ 1179 3038 ]]  @jdk/incubator/vector/Byte256Vector$Byte256Mask:exact+16 *, name=mfield, idx=10; mismatched #vectory[32]:{bool} !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
   51  ConP  === 0  [[ 52 60 137 146 252 259 332 340 452 434 557 565 801 788 2286 1058 1066 1081 1179 2286 3018 1808 1823 1907 1915 1969 1956 2279 2969 2994 3030 3031 3034 3036 3036 ]]  #NULL
    0  Root  === 0 64 83 150 166 263 280 344 360 438 456 569 597 620 633 646 684 696 708 792 805 1070 1090 1188 1812 1832 1919 1960 1973 2243 2255 2267 2290 2305 2365 2394 2463 2834 2557 2833 2627 2826 2669 2813 2731 2756  |2152 2172 2504  [[ 0 1 3 2487 23 2435 26 2349 2343 32 34 50 51 59 69 78 2205 107 134 145 2155 2154 601 2821 329 1914 1771 1166 411 413 433 451 478 1099 500 502 1036 935 506 858 641 628 534 554 615 ]] 
 2876  IfFalse  === 2875  [[ 1181 1177 3002 ]] #0 !orig=[1175] !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  668  If  === 652 667  [[ 669 670 ]] P=0.010156, C=23040.000000 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
 2162  If  === 2152 2161  [[ 2163 2164 ]] P=0.989844, C=22806.000000 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
  562  CastPP  === 561 2970  [[ 593 570 587 ]]  #jdk/internal/vm/vector/VectorSupport$VectorPayload:NotNull *  Oop:jdk/internal/vm/vector/VectorSupport$VectorPayload:NotNull * !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 2884  IfFalse  === 2883  [[ 587 611 3041 ]] #0 !orig=[585] !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1179  InlineType  === _ 51 34 1181  [[ 1178 ]]  #jdk/internal/vm/vector/VectorSupport$VectorPayloadMF256Z:NotNull:exact *  Oop:jdk/internal/vm/vector/VectorSupport$VectorPayloadMF256Z:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
   34  ConI  === 0  [[ 2503 2257 2245 2233 2171 1823 1822 1179 1178 1081 1080 698 609 622 635 674 686 3019 3018 2969 2970 2994 2995 1801 3033 3037 3042 ]]  #int:1
 1177  CheckCastPP  === 2876 587  [[ 1178 1180 1180 3034 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  669  IfTrue  === 668  [[ 673 ]] #1 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
 2164  IfFalse  === 2162  [[ 673 ]] #0 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
  587  CheckCastPP  === 2884 562  [[ 1828 1808 3002 1184 616 629 642 1177 2170 680 692 704 1169 1169 788 801 2978 1066 1086 1184 ]]  #jdk/incubator/vector/VectorMask:NotNull *  Oop:jdk/incubator/vector/VectorMask:NotNull * !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1178  InlineType  === _ 1177 34 1179  [[ 2152 2170 3002 1969 1956 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  673  Region  === 673 2164 669  [[ 673 2176 2167 2168 2893 2170 2172 3034 3036 3037 3038 ]]  !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:132 (line 5258)
 2170  Phi  === 673 1178 587  [[ ]]  #jdk/incubator/vector/VectorMask:NotNull *  Oop:jdk/incubator/vector/VectorMask:NotNull * !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
```
Can you kindly check why is InlineTypeNode not getting inserted before the second input(587) of phi node

Reply:
Ideally we should see following shape

![image](https://github.com/openjdk/valhalla/assets/59989778/5b3ebdb8-5be2-4774-9097-b450b7b00e26)

Reply:
> > Thanks, did not do a test re-run with latest patch as there were several regressions in pervious one, one quick comment, kindly set can_optimize = false before following line,
> > https://github.com/openjdk/valhalla/pull/845/files#diff-e368502932198721c3ee52ace2397c5a9c6281930e83fc53a429573f6adf44fcR2584
> 
> It seems not right setting `can_optimize = false` before line-2584 for me. This will break the inner loop anyway, since it doesn't set `can_optimize = true` in the followed code, does it?

I am seeing an incorrect graph shape here for ABSMaskedByte256VectorTests

```
 2970  VectorBox  === _ 2967 34 2969  [[ 562 557 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !orig=[3040],... !jvms: VectorMask::fromArray @ bci:47 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
  561  IfTrue  === 559  [[ 2883 562 ]] #1 !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 2883  If  === 561 2882  [[ 2884 2885 ]] P=0.000001, C=-1.000000 !orig=[584] !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1181  LoadVector  === 2876 2991 1180  [[ 1179 3038 ]]  @jdk/incubator/vector/Byte256Vector$Byte256Mask:exact+16 *, name=mfield, idx=10; mismatched #vectory[32]:{bool} !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
   51  ConP  === 0  [[ 52 60 137 146 252 259 332 340 452 434 557 565 801 788 2286 1058 1066 1081 1179 2286 3018 1808 1823 1907 1915 1969 1956 2279 2969 2994 3030 3031 3034 3036 3036 ]]  #NULL
    0  Root  === 0 64 83 150 166 263 280 344 360 438 456 569 597 620 633 646 684 696 708 792 805 1070 1090 1188 1812 1832 1919 1960 1973 2243 2255 2267 2290 2305 2365 2394 2463 2834 2557 2833 2627 2826 2669 2813 2731 2756  |2152 2172 2504  [[ 0 1 3 2487 23 2435 26 2349 2343 32 34 50 51 59 69 78 2205 107 134 145 2155 2154 601 2821 329 1914 1771 1166 411 413 433 451 478 1099 500 502 1036 935 506 858 641 628 534 554 615 ]] 
 2876  IfFalse  === 2875  [[ 1181 1177 3002 ]] #0 !orig=[1175] !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  668  If  === 652 667  [[ 669 670 ]] P=0.010156, C=23040.000000 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
 2162  If  === 2152 2161  [[ 2163 2164 ]] P=0.989844, C=22806.000000 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
  562  CastPP  === 561 2970  [[ 593 570 587 ]]  #jdk/internal/vm/vector/VectorSupport$VectorPayload:NotNull *  Oop:jdk/internal/vm/vector/VectorSupport$VectorPayload:NotNull * !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 2884  IfFalse  === 2883  [[ 587 611 3041 ]] #0 !orig=[585] !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1179  InlineType  === _ 51 34 1181  [[ 1178 ]]  #jdk/internal/vm/vector/VectorSupport$VectorPayloadMF256Z:NotNull:exact *  Oop:jdk/internal/vm/vector/VectorSupport$VectorPayloadMF256Z:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
   34  ConI  === 0  [[ 2503 2257 2245 2233 2171 1823 1822 1179 1178 1081 1080 698 609 622 635 674 686 3019 3018 2969 2970 2994 2995 1801 3033 3037 3042 ]]  #int:1
 1177  CheckCastPP  === 2876 587  [[ 1178 1180 1180 3034 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  669  IfTrue  === 668  [[ 673 ]] #1 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
 2164  IfFalse  === 2162  [[ 673 ]] #0 !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
  587  CheckCastPP  === 2884 562  [[ 1828 1808 3002 1184 616 629 642 1177 2170 680 692 704 1169 1169 788 801 2978 1066 1086 1184 ]]  #jdk/incubator/vector/VectorMask:NotNull *  Oop:jdk/incubator/vector/VectorMask:NotNull * !jvms: VectorMask::fromArray @ bci:50 (line 212) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:63 (line 5256)
 1178  InlineType  === _ 1177 34 1179  [[ 2152 2170 3002 1969 1956 ]]  #jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact *  Oop:jdk/incubator/vector/Byte256Vector$Byte256Mask:NotNull:exact * !jvms: Byte256Vector::lanewise @ bci:5 (line 279) Byte256Vector::lanewise @ bci:3 (line 42) Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:107 (line 5261)
  673  Region  === 673 2164 669  [[ 673 2176 2167 2168 2893 2170 2172 3034 3036 3037 3038 ]]  !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:132 (line 5258)
 2170  Phi  === 673 1178 587  [[ ]]  #jdk/incubator/vector/VectorMask:NotNull *  Oop:jdk/incubator/vector/VectorMask:NotNull * !jvms: Byte256VectorTests::ABSMaskedByte256VectorTests @ bci:86 (line 5259)
```

Can you check why are we injecting InlineTypeNode in first (Byte256Mask) input of  Phi, when phi itself is an oop. I suspect this could be an artifact of moving functionality into AbstractMask.java

Reply:
Phi is still carrying an abstract VectorMask type where as one of its input is concrete mask which is why C2 injects InlineTypeNode for it.

Reply:
Yes, so do you know how this PhiNode is generated and what it is used for? From the log, I can see the two inputs of Phi are all from `VectorBox`. I guess the abstract VectorMask is from the return type of `VectorMask.fromArray()` ?

Reply:
I also met several issues due to the VectorAPI abstract classes and its concrete vector classes. As I see, in compiler, sometimes the argument/return type is calculated from the sigunature type of the method, but sometimes it is calculated based on the speculative type which use the profiling info as a reference.  Following is part of my fixing for such cases:
```
diff --git a/src/hotspot/share/opto/callGenerator.cpp b/src/hotspot/share/opto/callGenerator.cpp
index 985f7f1c8..b77aa11d9 100644
--- a/src/hotspot/share/opto/callGenerator.cpp
+++ b/src/hotspot/share/opto/callGenerator.cpp
@@ -1169,7 +1169,7 @@ static void cast_argument(int nargs, int arg_nb, ciType* t, GraphKit& kit, bool
     kit.set_argument(arg_nb, arg);
   }
   if (sig_type->is_inlinetypeptr()) {
-    arg = InlineTypeNode::make_from_oop(&kit, arg, t->as_inline_klass(), !kit.gvn().type(arg)->maybe_null());
+    arg = InlineTypeNode::make_from_oop(&kit, arg, sig_type->inline_klass(), !kit.gvn().type(arg)->maybe_null());
     kit.set_argument(arg_nb, arg);
   }
 }
diff --git a/src/hotspot/share/opto/graphKit.cpp b/src/hotspot/share/opto/graphKit.cpp
index bce8d7781..bab94b204 100644
--- a/src/hotspot/share/opto/graphKit.cpp
+++ b/src/hotspot/share/opto/graphKit.cpp
@@ -1914,20 +1914,25 @@ Node* GraphKit::set_results_for_java_call(CallJavaNode* call, bool separate_io_p
   }
 
   // Capture the return value, if any.
-  Node* ret;
   if (call->method() == NULL || call->method()->return_type()->basic_type() == T_VOID) {
-    ret = top();
+    return top();
   } else if (call->tf()->returns_inline_type_as_fields()) {
     // Return of multiple values (inline type fields): we create a
     // InlineType node, each field is a projection from the call.
     ciInlineKlass* vk = call->method()->return_type()->as_inline_klass();
     uint base_input = TypeFunc::Parms;
-    ret = InlineTypeNode::make_from_multi(this, call, vk, base_input, false, call->method()->signature()->returns_null_free_inline_type());
+    return InlineTypeNode::make_from_multi(this, call, vk, base_input, false, call->method()->signature()->returns_null_free_inline_type());
   } else {
-    ret = _gvn.transform(new ProjNode(call, TypeFunc::Parms));
+    Node* ret = _gvn.transform(new ProjNode(call, TypeFunc::Parms));
+    ciType* t = call->method()->return_type();
+    if (t->is_klass()) {
+      const Type* sig_type = TypeOopPtr::make_from_klass(t->as_klass());
+      if (sig_type->is_inlinetypeptr()) {
+        ret = InlineTypeNode::make_from_oop(this, ret, sig_type->inline_klass(), false);
+      }
+    }
+    return ret;
   }
-
-  return ret;
 }
 
 //--------------------set_predefined_input_for_runtime_call--------------------
```

Reply:
> Yes, so do you know how this PhiNode is generated and what it is used for? From the log, I can see the two inputs of Phi are all from `VectorBox`. I guess the abstract VectorMask is from the return type of `VectorMask.fromArray()` ?

Do not consider empty output list of the PhiNode, this dump was taken from debugger during a reverse debugging step, so it may appear its not getting used. 

All the signatures of top level VectorAPIs methods always deal in abstract types, its only due to aggressive in-lining boxes (concrete vector/mask/shuffle) gets exposed. 

We have currently disabled value scalarization for vectors, please also add mask/shuffle to [that list](https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/prims/vectorSupport.cpp#L88)

https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/oops/inlineKlass.cpp#L341
https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/oops/inlineKlass.cpp#L336

Reply:
Have added by extending `is_vector()` like before. Please see: https://github.com/openjdk/valhalla/pull/845/files